src/main/kotlin/network/warzone/mars/player/commands/MiscCommands.kt

```diff
@@ -59,4 +64,44 @@ class MiscCommands {
         if (player == sender) sender.sendMessage("${ChatColor.GRAY}Your ping is ${ChatColor.AQUA}${ping}ms")
         else sender.sendMessage("${ChatColor.AQUA}$possessive ${ChatColor.GRAY}ping is ${ChatColor.AQUA}${ping}ms")
     }
-}
\ No newline at end of file
+
+    @Command(
+        aliases = ["tp"],
+        desc = "Teleport to a player",
+        usage = "<player> [otherPlayer]",
+        perms = ["mars.tp", "mars.tp.other"])
+    fun onTeleportToPlayer(
+        @Sender sender: Player,
+        @Nullable @PlayerName playerOneName: String?,
+        @Nullable @PlayerName playerTwoName: String?)
+    {
+        val targetPlayerOne = playerOneName?.let { Bukkit.getPlayer(it) }
+        val targetPlayerTwo = playerTwoName?.let { Bukkit.getPlayer(it) }
+
+        // Prevent even mods from using command if they are participating
+        // in the match. Opped players do not apply.
+        if (sender.matchPlayer.isParticipating && !sender.isOp) {
+            sender.sendMessage("You cannot use this command while participating!")
+            return
+        }
+        // Scenario One: Both arguments are empty.
+        if (targetPlayerOne == null) {
+            sender.sendMessage("You must provide a player to teleport to!")
+            return
+        }
+        // Scenario Two: Only argument two is empty.
+        if (targetPlayerTwo == null) {
+            sender.teleport(targetPlayerOne.location)
+            sender.sendMessage("Teleported to " + playerOneName)
+            return
+        }
+        // Scenario Three: Both arguments are filled.
+        if (sender.hasPermission("mars.tp.other")) {
+            targetPlayerOne.teleport(targetPlayerTwo.location)
+            sender.sendMessage("Teleported " + playerOneName + " to " + playerTwoName)
+            return
+        }
+
+        sender.sendMessage("If you see this message, something has gone wrong.")
```

Comment:
This will send the `If you see this message, something has gone wrong.` message to players without the `mars.tp.other` permission. Please change it to check if the player **doesn't** have the required permission to send the error message first and then the teleport method at the end since you don't have to check any other conditions.